cpp/include/algorithms.h

```diff
@@ -170,3 +172,20 @@ gdf_error gdf_louvain(gdf_graph *graph,
                       void *final_modularity,
                       void *num_level,
                       gdf_column *louvain_parts);
+
+/**
+ * @brief Compute connected components. 
+ * The weak version was imported from cuML.
+ * This implementation comes from [1] and solves component labeling problem in
+ * parallel on CSR-indexes based upon the vertex degree and adjacency graph.
+ *
+ * [1] Hawick, K.A et al, 2010. "Parallel graph component labelling with GPUs and CUDA"
+ *
+ 
+ * @param graph input graph; assumed undirected for weakly CC [in]
+ * @param labels gdf_column for the output labels [out]
+ * @param connectivity_type 0=WEAK; 1=STRONG
```

Comment:
If we ask users to give 0 for weakly connected component and 1 for strongly connected component, why did we define CUGRAPH_WEAK and CUGRAPH_STRONG in the above public header? Shouldn't we better say CUGRAPH_WEAK for weakly connected component and CUGRAPH_STRONG for strongly connected component to encourage users to use CUGRAPH_WEAK or CUGRAPH_STRONG instead of 0 or 1?

Reply:
Because in the Python binding these arguments should be passed as ints, probably. In which case the user must be warned of both. Enums resolve to ints, so I don't see what the issue is.

Reply:
Cython supports C++ enums, so if you're expecting cugraph_connect_t to expand over time and number assigned to CUGRAPH_WEAK or CUGRAPH_STRONG to change, we may better design cugraph's C++ public API to take enums instead of integer values (and encourage users to provide enum constatnts instead of integer values). If you expect we will consider only to cases (strong and weak), why not just use bool?

And python level, we need to follow NetworkX's convention. See the following link for NetworkX's API.

https://networkx.github.io/documentation/stable/reference/algorithms/component.html

So at the python level, users will not pass 0 or 1, but users will use different function names for weakly connected component and strongly connected component.